client/landing/stepper/declarative-flow/plugin-bundle-data.ts

```diff
@@ -1,26 +1,100 @@
-		{ slug: 'error', component: ErrorStep },
-	],
-};
+import type { StepperStep, Navigate } from './internals/types';
+import type { CalypsoDispatch } from 'calypso/state/types';
+
+/**
+ * First steps that will always run, regardless of the plugin bundle being registered here or not.
+ */
+export const initialBundleSteps: StepperStep[] = [
+	{
+		slug: 'getCurrentThemeSoftwareSets',
+		component: GetCurrentThemeSoftwareSets,
+	},
+];
+
+/**
+ * Steps that will run before the custom bundle steps.
+ */
+export const beforeCustomBundleSteps: StepperStep[] = [
+	{ slug: 'checkForPlugins', component: CheckForPlugins },
+];
+
+/**
+ * Steps that will run after the custom bundle steps.
+ */
+export const afterCustomBundleSteps: StepperStep[] = [
+	{ slug: 'bundleConfirm', component: BundleConfirm },
+	{ slug: 'bundleTransfer', component: BundleTransfer },
+	{ slug: 'bundleInstallPlugins', component: BundleInstallPlugins },
+	{ slug: 'processing', component: ProcessingStep },
+	{ slug: 'error', component: ErrorStep },
+];
+
+interface BundleStepsSettings {
+	[ key: string ]: {
+		/** Custom steps for the bundle. Empty string if it has not custom steps. */
+		customSteps: StepperStep[];
+		/** Customize back function only for custom steps of the flow. The default steps have their logic separately. It returns `false` if nothing should be done here. */
+		goBack?: ( currentStep: string, navigate: Navigate< StepperStep[] > ) => boolean | void;
+		/** Custom end of flow. Notice that it can end earlier depending on the current state. It returns `false` if nothing should be done here. */
+		endFlow?: ( {
+			intent,
+			storeType,
+			adminUrl,
+			dispatch,
+			exitFlow,
+		}: {
+			intent: string;
+			storeType: string;
+			adminUrl: string;
+			dispatch: CalypsoDispatch;
+			exitFlow: ( to: string ) => void;
+		} ) => boolean | void;
+		checkForActivePlugins: string[];
+	};
+}
 
-export type BundledPlugin = keyof typeof pluginBundleSteps;
+/**
+ * Steps settings for each plugin bundle.
+ */
+export const bundleStepsSettings: BundleStepsSettings = {
+	'woo-on-plans': {
+		customSteps: [
+			{ slug: 'storeAddress', component: StoreAddress },
+			{ slug: 'businessInfo', component: BusinessInfo },
+		],
+		goBack: ( currentStep, navigate ) => {
+			switch ( currentStep ) {
+				case 'businessInfo':
+					return navigate( 'storeAddress' );
+
+				case 'bundleConfirm':
+					return navigate( 'businessInfo' );
+
+				default:
+					return false;
+			}
+		},
+		endFlow: ( { intent, storeType, adminUrl, dispatch, exitFlow } ) => {
+			if ( intent === 'sell' && storeType === 'power' ) {
+				dispatch( recordTracksEvent( 'calypso_woocommerce_dashboard_redirect' ) );
+
+				return exitFlow( `${ adminUrl }admin.php?page=wc-admin` );
+			}
+
+			return false;
+		},
+		checkForActivePlugins: [ 'woocommerce' ],
+	},
+};
 
-export default pluginBundleSteps;
+export type BundledPlugin = keyof typeof bundleStepsSettings & string;
```

Comment:
I don't have much experience with TypeScript, so I'd appreciate help here if any reviewer has a suggestion.
`BundledPlugin` type is accepting any `string`.

If I remove the type `BundleStepsSettings` from the `bundleStepsSettings` it works properly, accepting only `"woo-on-plans"`. The same happens with the previous version of this object (`trunk`).

Any idea on how to solve it? The best resource I could find about this is https://stackoverflow.com/questions/51808160/keyof-inferring-string-number-when-key-is-only-a-string